lib/gridfs-stream/download.js

```diff
@@ -234,13 +234,12 @@ function doRead(_this) {
       _this.s.bytesToSkip = 0;
     }
 
-    if (expectedN === _this.s.expectedEnd && _this.s.bytesToTrim != null) {
-      sliceEnd = _this.s.bytesToTrim;
-    }
-
-    // If the remaining amount of data left is < chunkSize read the right amount of data
-    if (_this.s.options.end && _this.s.options.end - _this.s.bytesToSkip < buf.length) {
-      sliceEnd = _this.s.options.end - _this.s.bytesToSkip;
+    if (expectedN === _this.s.expectedEnd - 1 && _this.s.bytesToTrim != null) {
```

Comment:
can you hoist some of these statements out to make the conditionals more readable? I just skimmed this twice and am still like "whattt's going on here"